src/lib/utils/LoggerManager.cpp

```diff
@@ -14,9 +38,11 @@
  *
  * @param level The log level to set for the logger.
  */
-void LoggerManager::setup_logger(std::shared_ptr<config::Config> config, spdlog::level::level_enum level) {
+void LoggerManager::setup_logger(std::shared_ptr<config::Config> config) {
   try {
-    spdlog::trace("Setting up logger with level: {}", spdlog::level::to_string_view(level));
+    // Read environment variable
+    spdlog::cfg::load_env_levels();
+    spdlog::level::level_enum level = get_spdlog_level_from_env();
```

Comment:
Maybe use spdlog::get_level() to get the global log level or if the level of single logger is required   logger->level()